nomad/resource_acl_auth_method.go

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceACLAuthMethod() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceACLAuthMethodCreate,
+		Update: resourceACLAuthMethodUpdate,
+		Delete: resourceACLAuthMethodDelete,
+		Read:   resourceACLAuthMethodRead,
+		Exists: resourceACLAuthMethodExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The identifier of the ACL Auth Method.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"type": {
+				Description: `ACL Auth Method SSO workflow type. Currently, the only supported type is "OIDC."`,
+				Required:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					api.ACLAuthMethodTypeOIDC,
+				}, false),
+			},
+			"token_locality": {
+				Description: `Defines whether the ACL Auth Method creates a local or global token when performing SSO login. This field must be set to either "local" or "global".`,
+				Required:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					api.ACLAuthMethodTokenLocalityGlobal,
+					api.ACLAuthMethodTokenLocalityLocal,
+				}, false),
+			},
```

Comment:
Is this equivalent to the ACL token [`global`](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs/resources/acl_token#global) attribute? It may be nice to keep the name consistent if that's the case.

Reply:
It is and this mirrors what we have in Nomad; seems wrong to change it here and rather keep the consistency with naming between the provider and Nomad API.